cache/lru_cache.go

```diff
@@ -0,0 +1,140 @@
+// (c) 2019-2020, Ava Labs, Inc. All rights reserved.
```

Comment:
obviously a little copy/pasta from avalanchego, but I don't want to couple the cache implementation between the two.

Reply:
IMO it is totally ok to import this direct (there are already MANY coupled dependencies to AvalancheGo).